### PR TITLE
unhardcode mx_roadblock

### PR DESCRIPTION
**Pull request by nornagon**
#### Summary
None

#### Purpose of change

Refactoring away from add_spawn per #3951.

### CODE REVIEW

**Comment by nornagon** [comment]
Merged in code from #62415 to unhardcode mx_roadblock.

**Comment by Night-Pryanik** [comment]
Why are replacing working generator with a broken one? This change will also render searchlights non-operable.

**Comment by nornagon** [comment]
tagging @GuardianDll who did the mapgen json, do you have thoughts about what @Night-Pryanik said above?

**Comment by gettingusedto** [comment]
Does adding the "portable generator" vehicle and a couple of small storage batteries suffice or is it just a little too much?

File: data/json/mapgen/map_extras/roadblock_mil.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        "----------------------  ",
+        "----------------------  ",
+        "----------------------  ",
+        "-----------t----------  ",
+        "----------------------  ",
+        "----bbbbbbb--bbbbbbb--  ",
+        "                        ",
+        "                        "
+      ],
+      "furniture": { "s": "f_sandbag_half", "g": "f_generator_broken", "b": "f_barricade_road" },
```
Comment (DSeyka) [inline data/json/mapgen/map_extras/roadblock_mil.json:33]: If the generator is broken, how do turrets and searchlights still work?
I thought ASRG was added as an explanation to that.

**Comment by GuardianDll** [comment]
The reason is "we have no working generator furniture in the game except asrg, and adding one is out of scope of this pr. If someone know how to make such generator, I'd be glad to add one

**Comment by Night-Pryanik** [comment]
The special "attack" of the searchlight is hardcoded and explicitly requires `Compact Advanced Sterling Radioisotope Generator` to work:
https://github.com/CleverRaven/Cataclysm-DDA/blob/b50f199d9b48572021bcf7fc9bf81ef8a7ce8dca/src/monattack.cpp#L3828-L3829
It won't work with any other type of generators.

**Comment by GuardianDll** [comment]
Well. then it means someone else need to replace this generators in the future
I'll turn the ASRG back then

File: data/json/mapgen/map_extras/roadblock.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        "----------------------  ",
+        "----------------------  ",
+        "----------------------  ",
+        "------------t---------  ",
+        "----------------------  ",
+        "----bbbbbbb--bbbbbbb--  ",
+        "                        ",
+        "                        "
+      ],
+      "furniture": { "s": "f_sandbag_half", "g": "f_generator_broken", "b": "f_barricade_road" },
```
Comment (GuardianDll) [inline data/json/mapgen/map_extras/roadblock.json:43]:
```suggestion
      "furniture": { "s": "f_sandbag_half", "g": "f_compact_ASRG", "b": "f_barricade_road" },
```

File: data/json/mapgen/map_extras/roadblock_mil.json (hunk 1 above)
Comment (GuardianDll) [inline data/json/mapgen/map_extras/roadblock_mil.json:33]:
```suggestion
      "furniture": { "s": "f_sandbag_half", "g": "f_compact_ASRG", "b": "f_barricade_road" },
```

File: data/json/mapgen/map_extras/roadblock.json (hunk 2 above)
Comment (Night-Pryanik) [inline data/json/mapgen/map_extras/roadblock.json:43]:
```suggestion
      "furniture": { "s": "f_sandbag_half", "g": "f_compact_ASRG_containment ", "b": "f_barricade_road" },
```

File: data/json/mapgen/map_extras/roadblock_mil.json (hunk 1 above)
Comment (Night-Pryanik) [inline data/json/mapgen/map_extras/roadblock_mil.json:33]:
```suggestion
      "furniture": { "s": "f_sandbag_half", "g": "f_compact_ASRG_containment ", "b": "f_barricade_road" },
```

**Comment by oosyrag** [comment]
Wouldn't it make more sense to change the searchlight ability to not require an asrg instead? 

The generator could be a part of the searchlight itself.
https://www.larsonelectronics.com/category/599/Light-Towers-with-Generators

**Comment by GuardianDll** [comment]
After 0.G

**Comment by I-am-Erk** [comment]
We definitely don't want ASRG to be this common, they're a pretty rare and expensive object.

**Comment by RenechCDDA** [comment]
Heya, with #62459 merged you should be able to go ahead with this by replacing the ASRG with `f_active_backup_generator`.

File: data/json/mapgen/map_extras/roadblock_mil.json (hunk 1 above)
Comment (GuardianDll) [inline data/json/mapgen/map_extras/roadblock_mil.json:33]:
```suggestion
      "furniture": { "s": "f_sandbag_half", "g": "f_active_backup_generator", "b": "f_barricade_road" },
```

File: data/json/mapgen/map_extras/roadblock.json (hunk 2 above)
Comment (GuardianDll) [inline data/json/mapgen/map_extras/roadblock.json:43]:
```suggestion
      "furniture": { "s": "f_sandbag_half", "g": "f_active_backup_generator", "b": "f_barricade_road" },
```

**Review by GuardianDll — COMMENTED** [review]
so as RenechCDDA says, we can use better thing that ASRG from now

**Comment by Night-Pryanik** [comment]
@nornagon could you please resolve conflicts?

**Comment by nornagon** [comment]
@Night-Pryanik done

**Comment by nornagon** [comment]
@Night-Pryanik Conflicts resolved again, mind giving this another look before we conflict again 😅

**Comment by Night-Pryanik** [comment]
Another conflict coming on.

**Comment by nornagon** [comment]
aargh 😓 fixed! looks like Dark Skies Above got removed?

**Comment by Night-Pryanik** [comment]
Yep.

**Pull request merged**